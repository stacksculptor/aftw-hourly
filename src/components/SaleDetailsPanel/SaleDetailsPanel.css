@import '../../marketplace.css';

.root {
  padding: 0 24px;
}

.nowrap {
  white-space: nowrap;
}

.header {
  display: flex;
  margin-top: 28px;
  margin-bottom: 24px;
}

.title {
  flex: 1;
  margin: 0 12px 0 0;

  @media (--viewportMedium) {
    margin: 0 12px 0 0;
  }
}

.message {
  margin: 0 0 47px 0;
}

.avatarWrapper {
  width: auto;
  margin-top: 7px;
}

.avatar span {
  margin-bottom: 5px;
}

.bookingBreakdownContainer {
  margin-bottom: 40px;
}

.bookingBreakdownTitle {
  /* Font */
  color: var(--matterColorAnti);

  margin-top: 5px;
  margin-bottom: 14px;
  @media (--viewportMedium) {
    margin-top: 5px;
    margin-bottom: 14px;
  }
}
