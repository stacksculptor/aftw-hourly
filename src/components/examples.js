/* eslint-disable import/prefer-default-export*/
import * as Hello from './Hello/Hello.example';

export { Hello };
