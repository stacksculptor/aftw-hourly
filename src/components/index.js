import AddImages from './AddImages/AddImages';
import AuthorInfo from './AuthorInfo/AuthorInfo';
import BookingInfo from './BookingInfo/BookingInfo';
import Button from './Button/Button';
import CurrencyInput from './CurrencyInput/CurrencyInput';
import DateInput from './DateInput/DateInput';
import Discussion from './Discussion/Discussion';
import FilterPanel from './FilterPanel/FilterPanel';
import HeroSection from './HeroSection/HeroSection';
import Input from './Input/Input';
import LabeledField from './LabeledField/LabeledField';
import ListingCard from './ListingCard/ListingCard';
import ListingCardSmall from './ListingCardSmall/ListingCardSmall';
import LocationAutocompleteInput from './LocationAutocompleteInput/LocationAutocompleteInput';
import Map from './Map/Map';
import MapPanel from './MapPanel/MapPanel';
import Menu from './Menu/Menu';
import MobileFrame from './MobileFrame/MobileFrame';
import ModalInMobile from './ModalInMobile/ModalInMobile';
import NamedLink from './NamedLink/NamedLink';
import NamedRedirect from './NamedRedirect/NamedRedirect';
import OrderDetailsPanel from './OrderDetailsPanel/OrderDetailsPanel';
import OrderDiscussionPanel from './OrderDiscussionPanel/OrderDiscussionPanel';
import PageLayout from './PageLayout/PageLayout';
import PaginationLinks from './PaginationLinks/PaginationLinks';
import Promised from './Promised/Promised';
import RoutesProvider from './RoutesProvider/RoutesProvider';
import SearchResultsPanel from './SearchResultsPanel/SearchResultsPanel';
import StripeBankAccountToken from './StripeBankAccountToken/StripeBankAccountToken';

export {
  AddImages,
  AuthorInfo,
  BookingInfo,
  Button,
  CurrencyInput,
  DateInput,
  Discussion,
  FilterPanel,
  HeroSection,
  Input,
  LabeledField,
  ListingCard,
  ListingCardSmall,
  LocationAutocompleteInput,
  Map,
  MapPanel,
  Menu,
  MobileFrame,
  ModalInMobile,
  NamedLink,
  NamedRedirect,
  OrderDetailsPanel,
  OrderDiscussionPanel,
  PageLayout,
  PaginationLinks,
  Promised,
  RoutesProvider,
  SearchResultsPanel,
  StripeBankAccountToken,
};
