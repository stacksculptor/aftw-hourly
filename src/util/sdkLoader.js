import * as importedSdk from 'sharetribe-sdk';

let exportSdk;

const isServer = () => typeof window === 'undefined';

if (isServer()) {
  // Use eval to skip webpack from bundling SDK in Node
  // eslint-disable-next-line no-eval
  exportSdk = eval('require')('sharetribe-sdk');
} else {
  exportSdk = importedSdk;
}

const {
  createInstance,
  types,
} = exportSdk;

export { createInstance, types };
