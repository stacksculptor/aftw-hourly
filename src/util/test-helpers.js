import React from 'react';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from '../store';

export const TestProvider = props => {
  const store = configureStore();
  return (
    <IntlProvider locale="en">
      <BrowserRouter>
        <Provider store={store}>
          {props.children}
        </Provider>
      </BrowserRouter>
    </IntlProvider>
  );
};
