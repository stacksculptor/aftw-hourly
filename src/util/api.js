import { types as sdkTypes, transit } from './sdkLoader';
import Decimal from 'decimal.js';

const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return '/';
};

const typeHandlers = [
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window
    .fetch(url, options)
    .then(res => {
      if (res.status >= 400) {
        const e = new Error('Local API request failed');
        e.apiResponse = res;
        throw e;
      }
      return res;
    })
    .then(res => {
      const contentTypeHeader = res.headers.get('Content-Type');
      const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;
      if (contentType === 'application/transit+json') {
        return res.text().then(deserialize);
      } else if (contentType === 'application/json') {
        return res.json();
      }
      return res.text();
    });
};

export const transactionLineItems = body => {
  return post('/api/transaction-line-items', body);
};

export const initiatePrivileged = body => {
  return post('/api/initiate-privileged', body);
};
