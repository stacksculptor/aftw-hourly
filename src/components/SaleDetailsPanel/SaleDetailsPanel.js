import React, { PropTypes } from 'react';
import { FormattedDate, FormattedMessage } from 'react-intl';
import * as propTypes from '../../util/propTypes';
import { types } from '../../util/sdkLoader';
import { createSlug } from '../../util/urlHelpers';
import { Avatar, BookingInfo, NamedLink } from '../../components';

import css from './SaleDetailsPanel.css';

const SaleDetailsPanel = props => {
  const {
    className,
    subtotalPrice,
    saleState,
    booking,
    lastTransitionedAt,
    listing,
    customer,
    commission,
  } = props;
  const { firstName, lastName } = customer.attributes.profile;
  const customerName = firstName ? `${firstName} ${lastName}` : '';

  const listingLinkParams = { id: listing.id.uuid, slug: createSlug(listing.attributes.title) };
  const listingLink = (
    <NamedLink name="ListingPage" params={listingLinkParams}>
      {listing.attributes.title}
    </NamedLink>
  );

  // TODO We can't use price from listing, since that might have changed.
  // When API includes unit price and possible additional fees, we need to change this.
  const unitPrice = listing.attributes.price;

  const bookingInfo = unitPrice
    ? <BookingInfo
        className={css.receipt}
        bookingStart={booking.attributes.start}
        bookingEnd={booking.attributes.end}
        unitPrice={unitPrice}
        commission={commission}
        subtotalPrice={subtotalPrice}
      />
    : <p className={css.error}>{'priceRequiredMessage'}</p>;

  // saleState affects to both title and message section
  let stateMsgData = {};
  switch (saleState) {
    case propTypes.TX_STATE_PREAUTHORIZED:
      stateMsgData = {
        title: (
          <FormattedMessage
            id="SaleDetailsPanel.listingRequestedTitle"
            values={{ customerName: customerName, title: listingLink }}
          />
        ),
        message: (
          <div className={css.message}>
            <FormattedMessage id="SaleDetailsPanel.saleRequestedStatus" values={{ customerName }} />
          </div>
        ),
      };
      break;
    case propTypes.TX_STATE_ACCEPTED:
      stateMsgData = {
        title: (
          <FormattedMessage
            id="SaleDetailsPanel.listingAcceptedTitle"
            values={{ customerName: customerName, title: listingLink }}
          />
        ),
        message: (
          <div className={css.message}>
            <FormattedMessage id="SaleDetailsPanel.saleAcceptedStatus" />
            <FormattedMessage
              id="SaleDetailsPanel.onDate"
              values={{
                formattedDate: (
                  <FormattedDate
                    value={lastTransitionedAt}
                    year="numeric"
                    month="short"
                    day="numeric"
                  />
                ),
              }}
            />
          </div>
        ),
      };
      break;
    case propTypes.TX_STATE_REJECTED:
      stateMsgData = {
        title: (
          <FormattedMessage
            id="SaleDetailsPanel.listingRejectedTitle"
            values={{ customerName: customerName, title: listingLink }}
          />
        ),
        message: (
          <div className={css.message}>
            <FormattedMessage id="SaleDetailsPanel.saleRejectedStatus" />
            <FormattedMessage
              id="SaleDetailsPanel.onDate"
              values={{
                formattedDate: (
                  <FormattedDate
                    value={lastTransitionedAt}
                    year="numeric"
                    month="short"
                    day="numeric"
                  />
                ),
              }}
            />
          </div>
        ),
      };
      break;
    case propTypes.TX_STATE_DELIVERED:
      stateMsgData = {
        title: (
          <FormattedMessage
            id="SaleDetailsPanel.listingDeliveredTitle"
            values={{ customerName: customerName, title: listingLink }}
          />
        ),
        message: (
          <div className={css.message}>
            <FormattedMessage id="SaleDetailsPanel.saleDeliveredStatus" />
            <FormattedMessage
              id="SaleDetailsPanel.onDate"
              values={{
                formattedDate: (
                  <FormattedDate
                    value={lastTransitionedAt}
                    year="numeric"
                    month="short"
                    day="numeric"
                  />
                ),
              }}
            />
          </div>
        ),
      };
      break;
    default:
      stateMsgData = { title: null, message: null };
  }

  return (
    <div className={className}>
      <div className={css.messagesContainer}>
        <div className={css.avatarWrapper}>
          <Avatar firstName={firstName} lastName={lastName} />
        </div>
        <h1 className={css.title}>
          {stateMsgData.title}
        </h1>
        {stateMsgData.message}
      </div>
      {bookingInfo}
    </div>
  );
};

SaleDetailsPanel.defaultProps = { className: null };

const { instanceOf, string } = PropTypes;

SaleDetailsPanel.propTypes = {
  className: string,
  subtotalPrice: instanceOf(types.Money).isRequired,
  commission: instanceOf(types.Money).isRequired,
  saleState: string.isRequired,
  lastTransitionedAt: instanceOf(Date).isRequired,
  booking: propTypes.booking.isRequired,
  listing: propTypes.listing.isRequired,
  customer: propTypes.user.isRequired,
};

export default SaleDetailsPanel;
