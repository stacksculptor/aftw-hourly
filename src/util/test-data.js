import Decimal from 'decimal.js';
import { types } from './sdkLoader';
import { nightsBetween } from '../util/dates';
import * as propTypes from '../util/propTypes';

const { UUID, LatLng, Money } = types;

// Create a booking that conforms to the util/propTypes booking schema
export const createBooking = (id, attributes = {}) => ({
  id: new UUID(id),
  type: 'booking',
  attributes: {
    start: new Date(Date.UTC(2017, 5, 10)),
    end: new Date(Date.UTC(2017, 5, 10)),
    ...attributes,
  },
});

// Create a user that conforms to the util/propTypes user schema
export const createUser = id => ({
  id: new UUID(id),
  type: 'user',
  attributes: {
    banned: false,
    profile: {
      displayName: `${id} display name`,
      abbreviatedName: `${id} abbreviated name`,
    },
  },
});

// Create a user that conforms to the util/propTypes currentUser schema
export const createCurrentUser = id => ({
  id: new UUID(id),
  type: 'current-user',
  attributes: {
    banned: false,
    email: `${id}@example.com`,
    emailVerified: true,
    profile: {
      firstName: `${id} first name`,
      lastName: `${id} last name`,
      displayName: `${id} display name`,
      abbreviatedName: `${id} abbreviated name`,
    },
    stripeConnected: true,
  },
});

// Create a user that conforms to the util/propTypes user schema
export const createImage = id => ({
  id: new UUID(id),
  type: 'image',
  attributes: {
    sizes: [
      {
        name: 'square',
        height: 408,
        width: 408,
        url: 'https://placehold.it/408x408',
      },
      {
        name: 'square2x',
        height: 816,
        width: 816,
        url: 'https://placehold.it/816x816',
      },
    ],
  },
});

// Create a user that conforms to the util/propTypes listing schema
export const createListing = (id, attributes = {}, includes = {}) => ({
  id: new UUID(id),
  type: 'listing',
  attributes: {
    title: `${id} title`,
    description: `${id} description`,
    address: `${id} address`,
    geolocation: new LatLng(40, 60),
    closed: false,
    deleted: false,
    price: new Money(5500, 'USD'),
    ...attributes,
  },
  ...includes,
});

export const createTransaction = options => {
  const {
    id,
    lastTransition = propTypes.TX_TRANSITION_PREAUTHORIZE,
    total = new Money(1000, 'USD'),
    commission = new Money(100, 'USD'),
    booking = null,
    listing = null,
    customer = null,
    provider = null,
    lastTransitionedAt = new Date(Date.UTC(2017, 5, 1)),
  } = options;
  const nightCount = booking ? nightsBetween(booking.attributes.start, booking.attributes.end) : 1;
  return {
    id: new UUID(id),
    type: 'transaction',
    attributes: {
      createdAt: new Date(Date.UTC(2017, 4, 1)),
      lastTransitionedAt,
      lastTransition,
      payinTotal: total,
      payoutTotal: new Money(total.amount - commission.amount, total.currency),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(nightCount),
          unitPrice: new Money(total.amount / nightCount, total.currency),
          lineTotal: total,
          reversal: false,
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(commission.amount * -1, commission.currency),
          lineTotal: new Money(commission.amount * -1, commission.currency),
          reversal: false,
        },
      ],
    },
    booking,
    listing,
    customer,
    provider,
  };
};

// Default config for currency formatting in tests and examples.
export const currencyConfig = {
  style: 'currency',
  currency: 'USD',
  currencyDisplay: 'symbol',
  useGrouping: true,
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
};

const pad = num => {
  return num >= 0 && num < 10 ? `0${num}` : `${num}`;
};

// Create fake Internalization object to help with shallow rendering.
export const fakeIntl = {
  formatDate: d => `${d.getUTCFullYear()}-${pad(d.getUTCMonth() + 1)}-${pad(d.getUTCDate())}`,
  formatHTMLMessage: d => d,
  formatMessage: msg => msg.id,
  formatNumber: d => d,
  formatPlural: d => d,
  formatRelative: d => d,
  formatTime: d => `${pad(d.getUTCHours())}:${pad(d.getUTCMinutes())}`,
  now: () => Date.UTC(2017, 10, 23, 12, 59),
};

const noop = () => null;

export const fakeFormProps = {
  anyTouched: false,
  asyncValidating: false,
  dirty: false,
  form: 'fakeTestForm',
  invalid: false,
  pristine: true,
  clearSubmit: noop,
  touch: noop,
  untouch: noop,
  submit: noop,
  reset: noop,
  initialize: noop,
  handleSubmit: noop,
  destroy: noop,
  clearAsyncError: noop,
  change: noop,
  blur: noop,
  autofill: noop,
  asyncValidate: noop,
  valid: true,
  submitSucceeded: false,
  submitFailed: false,
  submitting: false,
  pure: true,
  initialized: true,
};
