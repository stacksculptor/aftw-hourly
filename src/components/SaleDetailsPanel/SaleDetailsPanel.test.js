import React from 'react';
import { shallow } from 'enzyme';
import { types } from '../../util/sdkLoader';
import { createTransaction, createBooking, createListing, createUser } from '../../util/test-data';
import { renderShallow } from '../../util/test-helpers';
import { BookingBreakdown } from '../../components';
import SaleDetailsPanel from './SaleDetailsPanel.js';

const noop = () => null;

describe('SaleDetailsPanel', () => {
  it('matches snapshot', () => {
    const { Money } = types;
    const transaction = createTransaction({
      id: 'sale-tx',
      state: 'state/preauthorized',
      total: new Money(16500, 'USD'),
      commission: new Money(1000, 'USD'),
      booking: createBooking(
        'booking1',
        new Date(Date.UTC(2017, 5, 10)),
        new Date(Date.UTC(2017, 5, 13))
      ),
      listing: createListing('listing1'),
      customer: createUser('customer1'),
      lastTransitionedAt: new Date(Date.UTC(2017, 5, 10)),
    });
    const props = {
      transaction,
      onAcceptSale: noop,
      onRejectSale: noop,
      acceptOrRejectInProgress: false,
    };
    const tree = renderShallow(<SaleDetailsPanel {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correct total price', () => {
    const { Money } = types;
    const transaction = createTransaction({
      id: 'sale-tx',
      state: 'state/preauthorized',
      total: new Money(16500, 'USD'),
      commission: new Money(1000, 'USD'),
      booking: createBooking(
        'booking1',
        new Date(Date.UTC(2017, 5, 10)),
        new Date(Date.UTC(2017, 5, 13))
      ),
      listing: createListing('listing1'),
      customer: createUser('customer1'),
      lastTransitionedAt: new Date(Date.UTC(2017, 5, 10)),
    });
    const props = {
      transaction,
      onAcceptSale: noop,
      onRejectSale: noop,
      acceptOrRejectInProgress: false,
    };
    const panel = shallow(<SaleDetailsPanel {...props} />);
    const breakdownProps = panel.find(BookingBreakdown).props();

    // Total price for the provider should be transaction total minus the commission.
    expect(breakdownProps.transaction.attributes.payoutTotal).toEqual(new Money(15500, 'USD'));
  });
});
