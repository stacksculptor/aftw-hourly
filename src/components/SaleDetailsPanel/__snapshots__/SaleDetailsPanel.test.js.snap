exports[`SaleDetailsPanel matches snapshot 1`] = `
<div
  className="">
  <div>
    <div>
      <ResponsiveImage
        alt="listing1 title"
        className={null}
        image={null}
        nameSet={
          Array [
            Object {
              "name": "landscape-crop",
              "size": "400w",
            },
            Object {
              "name": "landscape-crop2x",
              "size": "800w",
            },
          ]
        }
        noImageMessage={null}
        rootClassName={null}
        sizes="100vw" />
    </div>
    <div>
      <AvatarMedium
        user={
          Object {
            "attributes": Object {
              "banned": false,
              "profile": Object {
                "abbreviatedName": "customer1 abbreviated name",
                "displayName": "customer1 display name",
              },
            },
            "id": UUID {
              "uuid": "customer1",
            },
            "type": "user",
          }
        } />
    </div>
    <div>
      <div>
        <AvatarLarge
          user={
            Object {
              "attributes": Object {
                "banned": false,
                "profile": Object {
                  "abbreviatedName": "customer1 abbreviated name",
                  "displayName": "customer1 display name",
                },
              },
              "id": UUID {
                "uuid": "customer1",
              },
              "type": "user",
            }
          } />
      </div>
      <h1>
        <FormattedMessage
          id="SaleDetailsPanel.listingRequestedTitle"
          values={
            Object {
              "customerName": "customer1 display name",
              "listingLink": <withFlattenedRoutes(withRouter(NamedLink))
                name="ListingPage"
                params={
                    Object {
                        "id": "listing1",
                        "slug": "listing1-title",
                      }
                }>
                listing1 title
            </withFlattenedRoutes(withRouter(NamedLink))>,
            }
          } />
      </h1>
      <p>
        <FormattedMessage
          id="SaleDetailsPanel.saleRequestedStatus"
          values={
            Object {
              "customerName": "customer1 display name",
            }
          } />
      </p>
      <div />
      <div>
        <div />
        <SecondaryButton
          disabled={false}
          inProgress={false}
          onClick={[Function]}>
          <FormattedMessage
            id="SalePage.rejectButton"
            values={Object {}} />
        </SecondaryButton>
        <PrimaryButton
          disabled={false}
          inProgress={false}
          onClick={[Function]}>
          <FormattedMessage
            id="SalePage.acceptButton"
            values={Object {}} />
        </PrimaryButton>
      </div>
    </div>
    <div>
      <h3>
        <FormattedMessage
          id="SaleDetailsPanel.bookingBreakdownTitle"
          values={Object {}} />
      </h3>
      <BookingBreakdown
        booking={
          Object {
            "attributes": Object {
              "end": 2017-06-13T00:00:00.000Z,
              "start": 2017-06-10T00:00:00.000Z,
            },
            "id": UUID {
              "uuid": "booking1",
            },
            "type": "booking",
          }
        }
        transaction={
          Object {
            "attributes": Object {
              "createdAt": 2017-05-01T00:00:00.000Z,
              "lastTransition": "transition/preauthorize",
              "lastTransitionedAt": 2017-06-10T00:00:00.000Z,
              "lineItems": Array [
                Object {
                  "code": "line-item/night",
                  "lineTotal": Money {
                    "amount": 16500,
                    "currency": "USD",
                  },
                  "quantity": "3",
                  "unitPrice": Money {
                    "amount": 5500,
                    "currency": "USD",
                  },
                },
                Object {
                  "code": "line-item/provider-commission",
                  "lineTotal": Money {
                    "amount": -1000,
                    "currency": "USD",
                  },
                  "unitPrice": Money {
                    "amount": -1000,
                    "currency": "USD",
                  },
                },
              ],
              "payinTotal": Money {
                "amount": 16500,
                "currency": "USD",
              },
              "payoutTotal": Money {
                "amount": 15500,
                "currency": "USD",
              },
              "state": "state/preauthorized",
            },
            "booking": Object {
              "attributes": Object {
                "end": 2017-06-13T00:00:00.000Z,
                "start": 2017-06-10T00:00:00.000Z,
              },
              "id": UUID {
                "uuid": "booking1",
              },
              "type": "booking",
            },
            "customer": Object {
              "attributes": Object {
                "banned": false,
                "profile": Object {
                  "abbreviatedName": "customer1 abbreviated name",
                  "displayName": "customer1 display name",
                },
              },
              "id": UUID {
                "uuid": "customer1",
              },
              "type": "user",
            },
            "id": UUID {
              "uuid": "sale-tx",
            },
            "listing": Object {
              "attributes": Object {
                "address": "listing1 address",
                "closed": false,
                "deleted": false,
                "description": "listing1 description",
                "geolocation": LatLng {
                  "lat": 40,
                  "lng": 60,
                },
                "price": Money {
                  "amount": 5500,
                  "currency": "USD",
                },
                "title": "listing1 title",
              },
              "id": UUID {
                "uuid": "listing1",
              },
              "type": "listing",
            },
            "provider": null,
            "type": "transaction",
          }
        }
        userRole="provider" />
    </div>
    <div>
      <div>
        <ResponsiveImage
          alt="listing1 title"
          className={null}
          image={null}
          nameSet={
            Array [
              Object {
                "name": "landscape-crop",
                "size": "400w",
              },
              Object {
                "name": "landscape-crop2x",
                "size": "800w",
              },
            ]
          }
          noImageMessage={null}
          rootClassName={null}
          sizes="100%" />
      </div>
      <h3>
        <FormattedMessage
          id="SaleDetailsPanel.bookingBreakdownTitle"
          values={Object {}} />
      </h3>
      <div>
        <BookingBreakdown
          booking={
            Object {
              "attributes": Object {
                "end": 2017-06-13T00:00:00.000Z,
                "start": 2017-06-10T00:00:00.000Z,
              },
              "id": UUID {
                "uuid": "booking1",
              },
              "type": "booking",
            }
          }
          transaction={
            Object {
              "attributes": Object {
                "createdAt": 2017-05-01T00:00:00.000Z,
                "lastTransition": "transition/preauthorize",
                "lastTransitionedAt": 2017-06-10T00:00:00.000Z,
                "lineItems": Array [
                  Object {
                    "code": "line-item/night",
                    "lineTotal": Money {
                      "amount": 16500,
                      "currency": "USD",
                    },
                    "quantity": "3",
                    "unitPrice": Money {
                      "amount": 5500,
                      "currency": "USD",
                    },
                  },
                  Object {
                    "code": "line-item/provider-commission",
                    "lineTotal": Money {
                      "amount": -1000,
                      "currency": "USD",
                    },
                    "unitPrice": Money {
                      "amount": -1000,
                      "currency": "USD",
                    },
                  },
                ],
                "payinTotal": Money {
                  "amount": 16500,
                  "currency": "USD",
                },
                "payoutTotal": Money {
                  "amount": 15500,
                  "currency": "USD",
                },
                "state": "state/preauthorized",
              },
              "booking": Object {
                "attributes": Object {
                  "end": 2017-06-13T00:00:00.000Z,
                  "start": 2017-06-10T00:00:00.000Z,
                },
                "id": UUID {
                  "uuid": "booking1",
                },
                "type": "booking",
              },
              "customer": Object {
                "attributes": Object {
                  "banned": false,
                  "profile": Object {
                    "abbreviatedName": "customer1 abbreviated name",
                    "displayName": "customer1 display name",
                  },
                },
                "id": UUID {
                  "uuid": "customer1",
                },
                "type": "user",
              },
              "id": UUID {
                "uuid": "sale-tx",
              },
              "listing": Object {
                "attributes": Object {
                  "address": "listing1 address",
                  "closed": false,
                  "deleted": false,
                  "description": "listing1 description",
                  "geolocation": LatLng {
                    "lat": 40,
                    "lng": 60,
                  },
                  "price": Money {
                    "amount": 5500,
                    "currency": "USD",
                  },
                  "title": "listing1 title",
                },
                "id": UUID {
                  "uuid": "listing1",
                },
                "type": "listing",
              },
              "provider": null,
              "type": "transaction",
            }
          }
          userRole="provider" />
      </div>
    </div>
  </div>
</div>
`;
